test/bayesian_linear_regression.jl

```diff
@@ -121,4 +121,27 @@ end
             @test cov(f′(X′, Σy)) ≈ cov(f′2(X′, Σy))
         end
     end
+    @testset "sampling functions" begin
+        rng, N, D = MersenneTwister(123456), 11, 5
+        X, f, Σy = generate_toy_problem(rng, N, D)
+
+        g = rand(rng, f)
+        @test g(X) == g(X)  # check the sample doesn't change between evaluations
+
+        Xc = ColVecs(X)
+        Xr = RowVecs(X')
+        @test g(X) == g(Xc)
+        @test g(X) == g(Xr)
+
+        samples1, samples2 = 10_000, 1000
+        gs = rand(rng, f, samples1, samples2)
```

Comment:
Actually, scratch the above, I've realised I don't understand what's going on here. What should I expect `gs` to be?

Reply:
Oh, wait, nvm. It produces a matrix of samples. I had literally no idea that was a thing that the distributions API permitted. Or this a std lib thing?

Reply:
`g = rand(rng, f)` returns a `BLRFunctionSample`
`gs = rand(rng, f, n, m, o)` returns an array of `BLRFunctionSample` where `size(gs) = (n, m, o)`

Reply:
That's the convention of the `Random` API which I tried to follow

Reply:
But I suppose since we need non-mutating rand using the `Sampler` interface is a bit pointless anyway

Reply:
> But I suppose since we need non-mutating rand using the Sampler interface is a bit pointless anyway

It's not obvious to me that this is true. In particular, the single-sample version of `rand` that you've implemented appears to me to be mutation-free. Is that not the case?